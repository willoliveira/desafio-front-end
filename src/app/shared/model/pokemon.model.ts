import { IRessource } from "./ressource.model";

export enum EnumPokemonTypes {
	'normal',
	'fighting',
	'flying',
	'poison',
	'ground',
	'rock',
	'bug',
	'ghost',
	'steel',
	'fire',
	'water',
	'grass',
	'electric',
	'psychic',
	'ice',
	'dragon',
	'dark',
	'fairy',
	'unknown',
	'shadow'
}

export interface IPokemonAbilities {
	ability: IRessource;
	is_hidden: boolean;
	slot: number;
}

export interface IPokemonStats {
	base_stat: number;
	effort: number;
	stat: IRessource;
}

export interface IPokemonTypes {
	slot: number;
	type: IRessource;
}

export interface IPokemonSprites {
	back_default: string;
	back_female: string;
	back_shiny: string;
	back_shiny_female: string;
	front_default: string;
	front_female: string;
	front_shiny: string;
	front_shiny_female: string;
}

export interface IPokemon {
	id: number;
	base_experience: number;
	name: string;
	sprites: IPokemonSprites;
	weight: number;
	height: number;
	types: IPokemonTypes[];
	abilities: IPokemonAbilities[];
	stats: IPokemonStats[];
}
