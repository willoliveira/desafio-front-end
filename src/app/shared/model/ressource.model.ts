export interface IRessource {
	name: string;
	url: string;
}
