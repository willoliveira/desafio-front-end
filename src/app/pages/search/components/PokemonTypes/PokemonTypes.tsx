import React, { FC } from 'react';
import { IPokemonTypes } from '../../../../shared/model/pokemon.model';
import { View, StyleSheet, Text } from 'react-native';

export interface IPokemonTypeProps {
	types: IPokemonTypes[];
}

export const PokemonType: FC<IPokemonTypeProps> = ({ types }: IPokemonTypeProps) => (
	<View style={styles.typesContainer}>
		{
			types.map((type: IPokemonTypes) => (
				<View key={`key-${type.type.name}`} style={{ ...styles.typeContainer, backgroundColor: colorsTranspile[type.type.name]}}>
					<Text style={styles.typeNameText}>{type.type.name}</Text>
				</View>
			))
		}
	</View>
)

const styles = StyleSheet.create({
	typesContainer: {
		flex: 1,
		flexDirection: 'row',
		flexWrap: 'wrap',
		marginBottom: 24
	},
	typesContainerTitle: {
		fontSize: 18
	},
	typeContainer: {
		borderRadius: 8,
		padding: 8,
		minWidth: 56,
		margin: 2,
		textAlign: "center"
	},
	typeNameText: {
		color: '#FFFFFF',
		textAlign: 'center'
	},
});

const colorsTranspile = {
	'normal': '#a4acaf',
	'fighting': '#3dc7ef',
	'flying': '#8e6feb',
	'poison': '#b97fc9',
	'ground': '#f7de3f',
	'rock': '#a38c21',
	'bug': '#729f3f',
	'ghost': '#7b62a3',
	'steel': '#9eb7b8',
	'fire': '#fd7d24',
	'water': '#4592c4',
	'grass': '#9bcc50',
	'electric': '#eed535',
	'psychic': '#f366b9',
	'ice': '#51c4e7',
	'dragon': '#53a4cf',
	'dark': '#707070',
	'fairy': '#fdb9e9',
	'unknown': '#000000',
	'shadow': '#000000'
}
