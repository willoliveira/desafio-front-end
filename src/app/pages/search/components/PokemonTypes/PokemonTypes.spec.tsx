/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow } from 'enzyme';
import { PokemonType } from './PokemonTypes';
import { IPokemonTypes } from '../../../../shared/model/pokemon.model';

describe('<PokemonTypes />', () => {

	describe('Rendering', () => {

		it('should match to snapshot - one type - electric', () => {
			const types: IPokemonTypes[] = [{
				slot: 1, type: { name: 'electric', url: '' }
			}];
			const component = shallow(<PokemonType types={types} />);
			expect(component).toMatchSnapshot("PokemonType snapshot");
		});

		it('should match to snapshot - two type - poison and ', () => {
			const types: IPokemonTypes[] = [
				{ slot: 1, type: { name: 'poison', url: '' } },
				{ slot: 2, type: { name: 'grass', url: '' } },
			];
			const component = shallow(<PokemonType types={types} />);
			expect(component).toMatchSnapshot("PokemonType snapshot");
		});
	});
});
