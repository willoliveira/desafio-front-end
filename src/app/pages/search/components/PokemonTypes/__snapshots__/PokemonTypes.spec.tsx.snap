// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonTypes /> Rendering should match to snapshot - one type - electric: PokemonType snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "flexWrap": "wrap",
      "marginBottom": 24,
    }
  }
>
  <View
    key="key-electric"
    style={
      Object {
        "backgroundColor": "#eed535",
        "borderRadius": 8,
        "margin": 2,
        "minWidth": 56,
        "padding": 8,
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "textAlign": "center",
        }
      }
    >
      electric
    </Text>
  </View>
</View>
`;

exports[`<PokemonTypes /> Rendering should match to snapshot - two type - poison and : PokemonType snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "flexWrap": "wrap",
      "marginBottom": 24,
    }
  }
>
  <View
    key="key-poison"
    style={
      Object {
        "backgroundColor": "#b97fc9",
        "borderRadius": 8,
        "margin": 2,
        "minWidth": 56,
        "padding": 8,
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "textAlign": "center",
        }
      }
    >
      poison
    </Text>
  </View>
  <View
    key="key-grass"
    style={
      Object {
        "backgroundColor": "#9bcc50",
        "borderRadius": 8,
        "margin": 2,
        "minWidth": 56,
        "padding": 8,
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "textAlign": "center",
        }
      }
    >
      grass
    </Text>
  </View>
</View>
`;
